### PR TITLE
raise execpetion on save! if valid? returns false completes #70

### PR DESCRIPTION
**Pull request by seancookr**
Needs Review. Open to using respond_to instead, but I don't know if I see the point.  Please discuss.

### CODE REVIEW

File: test/document_test.rb [hunk 1]
```diff
@@ -395,6 +359,13 @@ def setup
     assert_not_equal foo.bar, bar.bar
   end
 
+  test 'subclass raises DocumentInvalidError if valid? fails' do
+    foo = ObjectWithValid.new
+    assert_raise Dolly::DocumentInvalidError do
+      foo.save!
+    end
+  end
```
Comment (netlogics) [inline test/document_test.rb]: I think an object named `ObjectWithValid` should pass validation?

**Comment by seancookr** [comment]
@netlogics what are your thoughts now?

**Comment by netlogics** [comment]
:+1:

**Comment by javierg** [comment]
I would leave the redactor on tests in a future PR, how you are doing it, you are hiding a lot and will make hard to understand what ía going on

**Comment by seancookr** [comment]
@javierg removed the refactor

**Comment by javierg** [comment]
:shipit:

**Pull request merged**